crates/sui-adapter-transactional-tests/tests/transfer_object/shared.exp

```diff
@@ -19,8 +19,8 @@ Version: 3
 Contents: test::m::S {id: sui::object::UID {id: sui::object::ID {bytes: fake(2,0)}}}
 
 task 4 'transfer-object'. lines 27-27:
-Error: Transaction Effects Status: Invalid command argument at 0. Immutable and shared objects cannot be passed by-value.
-Cannot return execution error with shared objects. Debug of error: CommandArgumentError { arg_idx: 0, kind: InvalidObjectByValue } at command Some(0)
+Error: Transaction Effects Status: Invalid Transfer Object, object does not have public transfer.
```

Comment:
I agree it would be better to have this error returned instead. To do so would involve adding a field to ObjectValue that indicates if the object is shared, do you think this update worth the footprint of making such a change?

Reply:
We can also get this kind of error by consolidating handling of shared object deletion to `apply_object_changes` (i.e. at the end of the transaction).